source/codegen/metadata/nixnetsocket/enums.py

```diff
@@ -1,18 +1,144 @@
 enums = {
-  'OptName': {
+  'OperationalStatus': {
+      'enum-value-prefix': 'OPERATIONAL_STATUS',
+        'values': [
+            {
+                'name': 'DOWN',
+                'value': 0
+            },
+            {
+                'name': 'UP',
+                'value': 1
+            }
+        ]
+    },
+     'IPStackInffoStringFormat': {
+        'enum-value-prefix': 'IPSTACK_INFO_STR_FORMAT',
+        'values': [
+            {
+                'name': 'JSON',
+                'value': 0
+            },
+            {
+                'name': 'TEXT',
+                'value': 1
+            }
+        ]
+    },
+    'AddressFamilies': {
+        'values': [
+            {
+                'name': 'AF_UNSPEC',
+                'value': 0
+            },
+            {
+                'name': 'AF_INET',
+                'value': 2
+            },
+            {
+                'name': 'AF_INET6',
+                'value': 10
+            }
+        ]
+    },
+    'ProtocolFamilies': {
+        'values': [
+            {
+                'name': 'PF_UNSPEC',
+                'value': 0
+            },
+            {
+                'name': 'PF_INET',
+                'value': 2
+            },
+            {
+                'name': 'PF_INET6',
+                'value': 10
+            }
+        ]
+    },
+    'IPProtocols': {
+        'values': [
+            {
+                'name': 'IPPROTO_IP',
+                'value': 0
+            },
+            {
+                'name': 'IPPROTO_TCP',
+                'value': 6
+            },
+            {
+                'name': 'IPPROTO_UDP',
+                'value': 8
+            },
+            {
+                'name': 'IPPROTO_IPV6',
+                'value': 12
+            }
+        ]
+    },
+    'SocketProtocolTypes': {
+        'values': [
+            {
+                'name': 'SOCK_STREAM',
+                'value': 1
+            },
+            {
+                'name': 'SOCK_DGRAM',
+                'value': 2
+            }
+        ]
+    },
+    'SocketOptionLevel': {
+        'values': [
+            {
+                'name': 'SOL_SOCKET',
+                'value': 13
+            }
+        ]
+    },
+    'OptName': {
         'values': [
             {
                 'name': 'TCP_NODELAY',
                 'value': 1
             },
             {
-                'name': 'IP_MULTICAST_TTL ',
+                'name': 'IP_ADD_MEMBERSHIP',
+                'value': 3
+            },
+            {
+                'name': 'IP_DROP_MEMBERSHIP',
+                'value': 4
+            },
+            {
+                'name': 'IP_MULTICAST_IF',
+                'value': 5
+            },
+            {
+                'name': 'IP_MULTICAST_TTL',
                 'value': 6
             },
+            {
+                'name': 'IPV6_JOIN_GROUP',
```

Comment:
Thanks for rounding these OptNames out, I think this is all of them from the header and have an email out to see if this is this is all the optnames or not.

I do think we should go ahead and add `IPV6_ADD_MEMBERSHIP` and `IPV6_DROP_MEMBERSHIP` with 12 and 13, respectively. It's duplicated with JOIN/LEAVE_GROUP but I think the codegen will properly add the allow_alias = true to allow the duplicate values in the proto Enum.

Reply:
I added these duplicate values and added the needed suppression to get allow_alias and build working